lib/Push.php

```diff
@@ -189,6 +189,30 @@ protected function encryptAndSign(Key $userKey, array $device, INotification $no
 		if ($isTalkNotification) {
 			$data['type'] = $notification->getObjectType();
 			$data['id'] = $notification->getObjectId();
+			if ($data['type'] === 'call') {
+				$richParameters = $notification->getRichSubjectParameters();
+				if (array_key_exists('call', $richParameters)) {
+					$call = $richParameters['call'];
+					switch ($call['call-type']) {
```

Comment:
Well afaik call type is sent as integer to apps anyway in all the endpoints, not as string. That's why the translation is done (and because of the missing pieces in the spreed app, which I've added in a PR there).

While I agree with the second point that this should be in the app, I had no idea how to extend that part properly.